src/pint/models/stand_alone_psr_binaries/BT_piecewise.py

```diff
@@ -0,0 +1,441 @@
+                group_no.append(self.piecewise_parameter_information[index_no][0])
+            else:
+                group_no.append(index_no)
+        # print(group_no)
+        return group_no
+
+    def get_group_boundaries(self):
+        lower_group_edge = []
+        upper_group_edge = []
+        # print("from get_group_boundaries")
+        if hasattr(self, "piecewise_parameter_information"):
+            for i in range(0, len(self.piecewise_parameter_information)):
+                lower_group_edge.append(
+                    self.piecewise_parameter_information[i][3].value
+                )
+                upper_group_edge.append(
+                    self.piecewise_parameter_information[i][4].value
+                )
+            return [lower_group_edge, upper_group_edge]
+
+    def a1(self):
+        self.A1_val = self.A1X_arr * ls
+        if hasattr(self, "A1X_per_toa") is True:
+            ret = self.A1X_per_toa + self.tt0 * self.A1DOT
+        else:
+            ret = self.A1 + self.tt0 * self.A1DOT
+        return ret
+
+    def get_tt0(self, barycentricTOA):
+        """tt0 = barycentricTOA - T0"""
+        if barycentricTOA is None or self.T0 is None:
+            tt0 = None
+            return tt0
+        if len(barycentricTOA) > 1:
+            if len(self.piecewise_parameter_information) > 0:
+                self.toa_belongs_in_group(
+                    barycentricTOA
+                )  # Defines object's index for each toa as an array of length = len(self._t)
+                (
+                    self.T0X_per_toa,
+                    self.A1X_per_toa,
+                ) = self.piecewise_parameter_from_information_array(
+                    self._t
+                )  # Uses the index for each toa array to create arrays where elements are the A1X/T0X to use with that toa
+        if len(barycentricTOA) > 1:
+            check = hasattr(self, "T0X_per_toa")
+            if hasattr(self, "T0X_per_toa") is True:
+                if len(self.T0X_per_toa) == 1:
+                    T0 = self.T0X_per_toa
+                    # print("hello from 1")
+                else:
+                    T0 = self.T0X_per_toa
+                    # print("hello from 2")
+            else:
+                T0 = self.T0
+                # print("hello from 3")
+        else:
+            T0 = self.T0
+            # print("hello from 4")
+        if not hasattr(barycentricTOA, "unit") or barycentricTOA.unit == None:
+            barycentricTOA = barycentricTOA * u.day
+        # print(f"Unique T0s being used in tt0 calculation: {np.unique(T0)}\n")
+        tt0 = (barycentricTOA - T0).to("second")
+        # print(f"piecewise tt0 being called, Calc'd with: {T0} ")
+        return tt0
+
+    def d_delayL1_d_par(self, par):
+        if par not in self.binary_params:
+            errorMesg = par + " is not in binary parameter list."
+            raise ValueError(errorMesg)
+        par_obj = getattr(self, par)
```

Comment:
Is this used anywhere?

Reply:
Can't tell the exact line this is referencing, but par_obj is used flexibly get the unit we expect with an arbitrary piecewise parameter.